src/main/java/core/basesyntax/services/FruitsServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.services;
+
+import core.basesyntax.dao.FruitsDao;
+import core.basesyntax.transactions.FruitsTransaction;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FruitsServiceImpl implements FruitsService {
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private final FruitsDao fruitsDao;
+
+    public FruitsServiceImpl(FruitsDao fruitsDao) {
+        this.fruitsDao = fruitsDao;
+    }
+
+    @Override
+    public void processTransactions(List<FruitsTransaction> transactions) {
+        transactions.stream().forEach(transaction -> {
+            switch (transaction.getOperation()) {
+                case BALANCE -> fruitsDao.balance(transaction.getFruit(),
+                        transaction.getQuantity());
+                case SUPPLY -> fruitsDao.supply(transaction.getFruit(),
+                        transaction.getQuantity());
+                case PURCHASE -> fruitsDao.purchase(transaction.getFruit(),
+                        transaction.getQuantity());
+                case RETURN -> fruitsDao.returnFruits(transaction.getFruit(),
+                        transaction.getQuantity());
+                default -> throw new IllegalArgumentException("Unknown operation"
+                        + transaction.getOperation());
+            }
+        });
```

Comment:
I would recommend switching from switch expression to a hashmap
With a hashmap, adding new strategies becomes as simple as adding a new key-value pair, without needing to alter the existing decision structure. This aligns well with the open-closed principle, which states that software entities should be open for extension but closed for modification.

A hashmap can help decouple the decision logic from the execution logic. Instead of having a switch-case which is tightly coupled with all the possible cases, you can have a hashmap which dynamically retrieves the strategy based on the operation.

For a large number of cases, a hashmap can offer better performance as it provides constant-time lookups (O(1)), whereas a switch-case might degenerate to a linear search (O(n)) in the worst-case scenario